core/trino-main/src/main/java/io/trino/operator/exchange/UniformPartitionRebalancer.java

```diff
@@ -386,30 +389,18 @@ public static long serialize(WriterPartitionId writerPartitionId)
             // contains partitionId.
             return ((long) writerPartitionId.writerId << 32 | writerPartitionId.partitionId & 0xFFFFFFFFL);
         }
-
-        public WriterPartitionId(int writerId, int partitionId)
-        {
-            this.writerId = writerId;
-            this.partitionId = partitionId;
-        }
     }
 
+    // TODO remove when error prone is updated for Java 17 records
+    @SuppressWarnings("unused")
     private record WriterId(int id)
     {
-        private WriterId(int id)
-        {
-            this.id = id;
-        }
     }
 
+    // TODO remove when error prone is updated for Java 17 records
+    @SuppressWarnings("unused")
```

Comment:
Let's not sprinkle these suppressions all over the place. Let's disable the error prone check until the issue gets fixed.

Reply:
Removed this commit for now